### PR TITLE
[Java] Add license to pom

### PR DESCRIPTION
**Pull request by cliffano**
Add license options (licenseName and licenseUrl) to Java-related pom.xml file.

Default license name and URL values are based on https://github.com/swagger-api/swagger-codegen/blob/master/LICENSE .

License is required for publishing to central repository https://maven.apache.org/guides/mini/guide-central-repository-upload.html .

### PR checklist

- [v] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [v] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [v] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

### CODE REVIEW

**Comment by cliffano** [comment]
Saw the conflicts, will sort it out.

**Comment by cliffano** [comment]
@wing328 How can I retrigger the Shippable and appveyor builds?
The conflict merge should pass `mvn clean package` now.

**Comment by wing328** [comment]
One way is to close and then reopen the PR.

**Comment by cliffano** [comment]
The Shippable failures occur on `git sync` step, doesn't seem to have anything to do with the actual Swagger Codegen build.
And for some reason, the conflicts are still there even though it's already resolve on java-pom-license branch.

**Comment by wing328** [comment]
No worry. I think I'll cherry-pick c6491dc855d62124c4dc6af97938fae91642dd6e

**Comment by cliffano** [comment]
@wing328 Thanks. Please let me know when it's picked.

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractJavaCodegen.java [hunk 1]
```diff
@@ -52,6 +52,8 @@
     protected String groupId = "io.swagger";
     protected String artifactId = "swagger-java";
     protected String artifactVersion = "1.0.0";
+    protected String licenseName = "Apache License, Version 2.0";
```
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractJavaCodegen.java]: I'll change this to `Unlicense` instead based on the discussion in https://github.com/swagger-api/swagger-codegen/issues/2963

**Comment by wing328** [comment]
Closed via #4380

**Comment by wing328** [comment]
@cliffano #4380 has been merged into master. Please pull the latest to give it a try. Thanks for the enhancement.

**Comment by cliffano** [comment]
@wing328 Thanks for that.

Btw, it seems like it's common for petstores to not get re-generated on each code change. Obviously, we are humans and do miss out on things.

Should there be a bot account that fetches latest and regenerates petstores? If there is no change, do nothing. Otherwise, submit a PR.

**Comment by wing328** [comment]
> The trouble is definitely when there are conflicts and/or test failures.

Definitely. Whenever we add a new test case, we encountered issues with updated Petstore sample for at least one language from time to time.

**Comment by cliffano** [comment]
@wing328 Are you fine with accepting bot-generated pull requests for petstore regeneration if there's no conflict and no test failure?

**Comment by wing328** [comment]
@cliffano good idea 👍  We can try with that.

**Pull request closed without merging**